tasks/add_picklist_values.py

```diff
@@ -0,0 +1,339 @@
+            field_namespace_query = "NamespacePrefix = '{}' AND ".format(
+                split_field[0]
+            )
+        else:
+            field_developer_name = split_field[0]
+            field_namespace_query = "NamespacePrefix = '' AND "
+
+        split_sobject = sobject.split('__')
+        if len(split_sobject) is 3:
+            sobject_developer_name = split_sobject[1]
+            sobject_namespace_query = "NamespacePrefix = '{}' AND ".format(
+                split_sobject[0]
+            )
+        else:
+            sobject_developer_name = split_sobject[0]
+            # Standard objects will not be found if NamespacePrefix is left in the query.
+            sobject_namespace_query = ""
+
+        sobject_tooling_results = self.tooling.query(
+            "SELECT DeveloperName, DurableId "
+            "FROM EntityDefinition "
+            "WHERE {namespace_query}"
+            "DeveloperName = '{sobject}'".format(
+                namespace_query=sobject_namespace_query,
+                sobject=sobject_developer_name
+            )
+        )
+
+        if sobject_tooling_results["size"] is not 1:
+            raise Exception('{} object not found'.format(sobject))
+
+        validated_sobject_id = sobject_tooling_results["records"][0]["DurableId"]
+
+        field_tooling_results = self.tooling.query(
+            "SELECT Id, DeveloperName, Metadata "
+            "FROM CustomField "
+            "WHERE {namespace_query}"
+            "DeveloperName = '{developer_name}' "
+            "AND TableEnumOrId = '{sobject_id}'".format(
+                namespace_query=field_namespace_query,
+                developer_name=field_developer_name,
+                sobject_id=validated_sobject_id
+            )
+        )
+
+        if field_tooling_results["size"] is not 1:
+            raise Exception('{} field not found on {}'.format(field, sobject))
+
+        validated_field = field_tooling_results["records"][0]["Metadata"]
+        existing_picklist_values = []
+
+        # build up a list of active record types
+        picklist_record_types = self._get_active_record_types(sobject)
+
+        if validated_field["type"] == "Picklist":
+            existing_picklist_values = validated_field["valueSet"]["valueSetDefinition"]["value"]
+        else:
+            raise Exception('{} field is not a picklist field'.format(field))
+
+        # build the package.xml:
+        package_xml = self._build_package_xml(picklist_record_types, sobject, field)
+
+        # build the object XML:
+        object_xml = self._build_object_xml(validated_field, picklist_record_types, existing_picklist_values, values, field)
+
+        # create temporary files & deploy back to Salesforce
+        self.tempdir = tempfile.mkdtemp()
+        package_xml_file = os.path.join(self.tempdir, "package.xml")
+        with open(package_xml_file, "w") as f:
+            f.write(package_xml)
+
+        os.mkdir("{}/{}".format(self.tempdir, "objects"))
```

Comment:
Does this work correctly in Windows? I would have thought you would need to join them with os.path.join to make sure it uses a backslash separator in Windows.

Reply:
@davisagli Yep, it's always worked for me. Though I'm using bash in VS Code terminal... not sure if that makes a difference...

Reply:
@robertbwatson I researched a bit. In Python 3.6+, Python is smart enough to treat the argument to os.mkdir as a [path-like object](https://docs.python.org/3/glossary.html#term-path-like-object) which will get turned into a real path for the filesystem in use as needed.